catalog/app/containers/Bucket/PackageDialog/FilesInput.tsx

```diff
@@ -1016,88 +1195,207 @@ export function FilesInput({
+    },
+    '& svg': {
+      fontSize: '18px',
+    },
+  },
+}))
+
+interface FilesSelectorEntry {
+  type: 'dir' | 'file'
+  name: string
+  selected: boolean
+  size?: number
+}
+
+export type FilesSelectorState = FilesSelectorEntry[]
+
+interface FilesSelectorProps {
+  input: {
+    value: FilesSelectorState
+    onChange: (value: FilesSelectorState) => void
+  }
+  className?: string
+  disabled?: boolean
+  errors?: Record<string, React.ReactNode>
+  meta: {
+    submitting: boolean
+    submitSucceeded: boolean
+    submitFailed: boolean
+    dirty: boolean
+    error?: string
+    initial: FilesSelectorState
+  }
+  title: React.ReactNode
+  truncated?: boolean
+}
+
+export function FilesSelector({
+  input: { value, onChange },
+  className,
+  disabled = false,
+  errors = {},
+  meta,
+  title,
+  truncated = false,
+}: FilesSelectorProps) {
+  const classes = useFilesSelectorStyles()
+
+  const submitting = meta.submitting || meta.submitSucceeded
+  const error = meta.submitFailed && meta.error
+
+  const selected = React.useMemo(
+    () => value.reduce((m, i) => (i.selected ? m + 1 : m), 0),
+    [value],
+  )
+
+  const toggleAll = React.useCallback(() => {
+    onChange(value.map(R.assoc('selected', selected < value.length)))
+  }, [onChange, value, selected])
+
+  const handleItemClick = React.useCallback(
+    (e: React.MouseEvent<HTMLDivElement>) => {
+      e.stopPropagation()
+      const idx = value.findIndex(R.propEq('name', e.currentTarget.dataset.name!))
+      if (idx < 0) return
+      onChange(R.adjust(idx, R.evolve({ selected: R.not }), value))
+    },
+    [onChange, value],
+  )
+
+  return (
+    <Root className={className}>
+      <Header>
+        <HeaderTitle
+          state={
+            submitting || disabled // eslint-disable-line no-nested-ternary
+              ? 'disabled'
+              : error // eslint-disable-line no-nested-ternary
+              ? 'error'
+              : truncated
+              ? 'warn'
+              : undefined
+          }
+        >
+          {title}
+          {truncated && (
+            // TODO: adjust copy
+            <M.Tooltip title="Only the first 1000 items are shown, but the folder contains more">
+              <M.Icon style={{ marginLeft: 6 }} fontSize="small">
+                error_outline
+              </M.Icon>
+            </M.Tooltip>
+          )}
+        </HeaderTitle>
+        <M.Box flexGrow={1} />
+        {value.length > 0 && (
+          <M.Button
+            onClick={toggleAll}
+            disabled={disabled || submitting}
+            size="small"
+            endIcon={
+              <M.Icon fontSize="small">
+                {selected < value.length ? 'check_box' : 'check_box_outline_blank'}
+              </M.Icon>
+            }
+          >
+            Select {selected < value.length ? 'all' : 'none'}
+          </M.Button>
+        )}
+      </Header>
+
+      <ContentsContainer>
+        <Contents error={!!error} warn={truncated}>
```

Comment:
I think it's better to rename `error` prop to something else, like `hasError` or `invalid`

Reply:
why? MUI uses similar convention and i don's see what's wrong with it

Reply:
It's harder to read and maintain code if identically named variables have different content

Reply:
well, they have similar meaning